test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/SolutionExplorer/Models/AssetsFileDependenciesSnapshotTests.cs

```diff
@@ -13,6 +13,4907 @@ namespace NuGet.VisualStudio.Implementation.Test.SolutionExplorer.Models
 {
     public class AssetsFileDependenciesSnapshotTests
     {
+        [Fact]
+        public void TransitiveDependencyWarningsPropagate()
+        {
+            #region Test data
+
+            var lockFileContent = """
```

Comment:
The tests might be easier to read if the assets file were embedded instead of hardcoded as strings, like: https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/compiler/resources/project.assets.json.

Reply:
I considered that but put it here so it's immediately next to the consumption.

Reply:
If the test data was less than one or two hundred lines, I could see it being useful next to the code that uses it. But at nearly 5000 lines, my personal opinion is that it's counterproductive, especially since it's JSON. As a string in-line, VS won't give me JSON navigation tools, like being able to collapse part of the document. In VSCode, the JSON view also shows the JSON Path to the cursor's location, which also aids in navigating large JSON files.